src/main/java/com/github/cameltooling/lsp/internal/references/ReferencesProcessor.java

```diff
@@ -99,16 +100,34 @@ private boolean isReference(CamelURIInstance camelURIInstanceToSearchReference,
 		return allCamelUriInstance;
 	}
 	
-	private String getDirectId(CamelURIInstance camelDirectURIInstance) {
+	private static String getDirectId(CamelURIInstance camelDirectURIInstance) {
 		Set<PathParamURIInstance> pathParams = camelDirectURIInstance.getComponentAndPathUriElementInstance().getPathParams();
 		if (!pathParams.isEmpty()) {
 			return pathParams.iterator().next().getValue();
 		}
 		return null;
 	}
 
-	private boolean isDirectComponentKind(CamelURIInstance camelURIInstanceToSearchReference) {
+	public static boolean isDirectComponentKind(CamelUriElementInstance camelURIInstanceToSearchReference) {
```

Comment:
if used in several places ReferencesProcessor.isDirectComponentKind doesn't make a lot of meaning to stay in ReferencesProcessor.
it would be better to extract it to a specific class

Reply:
relocated to util class